File: drop/wwwroot/js/imageController.js
(function () {
    "use strict";

    angular.module("images")
        .controller("imageController", imageController);

    function imageController($http) {

        var vm = this;
        vm.rowLimit = 4;

        vm.images = [];
        vm.newImage = {};

        vm.addImage = function () {
            $http.post("/image/upload", vm.newImage)
                .then(function(response) {
                    vm.images.push(response.data);
                    vm.newImage = {};
                }).finally(function() {
                    //busy false
                });

        }
    }
})();

File: drop/Controllers/ImageController.cs
using AutoMapper;
using drop.Models.Entity;
using drop.Models.ViewModels;
using Microsoft.AspNetCore.Mvc;

namespace drop.Controllers
{
    [Route("image/upload")]
    public class ImageController : Controller
    {
        [HttpPost]
        public IActionResult Post([FromBody]ImageViewModel image)
        {
            if (ModelState.IsValid)
            {
                var newImage = Mapper.Map<Image>(image);

                return Ok(true);
            }
            return BadRequest("Bad data");
        }
    }
}